test/MintBurnERC20.test.js

```diff
@@ -0,0 +1,176 @@
+const { expect } = require("chai");
+const { ethers, waffle, network } = require("hardhat");
+const { provider, deployContract } = waffle;
+
+const ERC20SafeContract = require("../artifacts/contracts/ERC20Safe.sol/ERC20Safe.json");
+const MintBurnERC20Contract = require("../artifacts/contracts/MintBurnERC20.sol/MintBurnERC20.json");
+const BridgeContract = require("../artifacts/contracts/Bridge.sol/Bridge.json");
+
+describe("ERC20Safe, MintBurnERC20, and Bridge Interaction", function () {
+  const [adminWallet, relayer1, relayer2, relayer3, relayer4, relayer5, relayer6, relayer7, relayer8, otherWallet] =
+    provider.getWallets();
+  const boardMembers = [
+    adminWallet,
+    relayer1,
+    relayer2,
+    relayer3,
+    relayer4,
+    relayer5,
+    relayer6,
+    relayer7,
+    relayer8,
+  ].map(m => m.address);
+  const quorum = 7;
+
+  let erc20Safe, bridge, mintBurnErc20;
+
+  async function setupContracts() {
+    erc20Safe = await deployContract(adminWallet, ERC20SafeContract);
+    bridge = await deployContract(adminWallet, BridgeContract, [boardMembers, quorum, erc20Safe.address]);
+    await erc20Safe.setBridge(bridge.address);
+    await bridge.unpause();
+    await setupErc20Token();
+  }
+
+  async function setupErc20Token() {
+    mintBurnErc20 = await deployContract(adminWallet, MintBurnERC20Contract, ["Test Token", "TST"]);
+    await erc20Safe.whitelistToken(mintBurnErc20.address, 0, 100, true);
+    await erc20Safe.unpause();
+  }
+
+  beforeEach(async function () {
+    await setupContracts();
+  });
+
+  describe("should burn tokens on deposit to ERC20Safe", async function () {
+    let amount, batchNonce, signatures;
+    let dataToSign, signature1, signature2, signature3, signature4, signature5, signature6, signature7;
+    beforeEach(async function () {
+      amount = 80;
+      batchNonce = 42;
+      signatures = await getSignaturesForExecuteTransfer(
+        [mintBurnErc20.address],
+        [otherWallet.address],
+        [amount],
+        [1],
+        batchNonce,
+      );
+    });
+
+    function getExecuteTransferData(tokenAddresses, recipientAddresses, amounts, depositNonces, batchNonce) {
```

Comment:
Can we refactor this code so it is reused? It's the same as in `Bridge.test.js`

Reply:
done